libtenzir/include/tenzir/concept/printable/tenzir/blob.hpp

```diff
@@ -0,0 +1,31 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The Tenzir Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+
+#include "tenzir/concept/printable/core.hpp"
+#include "tenzir/concept/printable/string/escape.hpp"
+#include "tenzir/detail/escapers.hpp"
+
+namespace tenzir {
+
+template <class T>
+struct generic_blob_printer : printer_base<generic_blob_printer<T>> {
```

Comment:
Does a `data` containing a `blob` survive a print-parse roundtrip?

Reply:
No, because `blob` currently does not have a parseable representation in TQL. This printer is just for internal purposes. Does this cause any issues?

Reply:
I don't think it does, but if it isn't much work it generally makes sense to add it to avoid surprises.

Reply:
Do we want to address this in a follow-up PR? We first have to agree on syntax, and then also have to address semantics in expressions (because it would then be surprising if these literals would misbehave in queries).

Reply:
Makes sense. Can you write one, mentioning also that the printer currently uses `b"…"`, and if we decide on another representation would have to change that as well?